### ISSUE TITLE
`break` and `continue` don't work in `for..let` loops and consequences of that (discussion)

### ISSUE THREAD

**Issue by pepkin88** [gkz/LiveScript#1024]
Because of how `for..let` loops wrap their bodies in the function, you cannot use `break` and `continue` in them. Hacking them to work, with labels, would be a bigger challenge, not impossible (Babel has it), but probably not worth the effort. That's why I believe that it is as intended, but I think this limitation should be documented.

However, one of the bummers in ES5 were, that you couldn't at the same time use `break` or `continue` and be having an iteration scoped variables, like in `Array.prototype.map`, at the same time. So the ES6 added `let`s and `const`s. But because in LS they are emulated using ES5 tricks, they come with ES5 limitations.
And perhaps there are more use cases, where JS `let`s make more sense, than LS helpers, though nothing comes to my mind at the moment.

And I also don't think it is super important. I would probably find a simple way to get around such issue in my code.

In summary, options "what to do":
- leave as it is, but document better
- implement them, but only on basic level
- implement them, but fully, with labels
- implement a fallback to a real `let`
- introduce a real `let` anyway, everywhere
- introduce a real `let`, but through Babel pipeline

I would opt for the first one or the Babel one, but that's another issue (#821?).

**Comment by mt4c** [comment]
also `return` do not break the loop and return a value
